### ISSUE TITLE
If SerilogTraceListerner is initialized before Log.Logger is set, the listener will not work

### ISSUE THREAD

**Issue by amat27** [serilog-trace-listener/SerilogTraceListener#24]
Sometimes in Release build, `SerilogTraceListener` will get initialized before `Log.Logger` can be set

**Comment by nblumhardt** [comment]
Thanks - yes, this is a bit of a gotcha. Currently it can be worked around by using code to configure the trace listener, and providing the logger there.

A PR fixing this would be welcome if anyone is interested in tackling it.

Thanks!

**Comment by jeoffman** [comment]
It is possible to overcome this if you create a new SerilogTraceListener for each TraceSource (as per below). Note you still have a problem for any other TraceSource which was used before the SerilogTraceListener was constructed by the system unless you use something like my HACK in PR #25 

```xml
    <source name="MyTraceSource" switchValue="All">
      <listeners>
        <clear />
        <add name="Serilog" type="SerilogTraceListener.SerilogTraceListener, SerilogTraceListener" />
      </listeners>
    </source>
```
Thanks to @amccool for this info

**Comment by nblumhardt** [comment]
Fixed in #25
